datadog_lambda/tracing.py

```diff
@@ -223,7 +223,24 @@ def extract_context_from_sqs_or_sns_event_or_context(event, lambda_context):
             first_record.get("Sns", {}).get("MessageAttributes", {}),
         )
         dd_payload = msg_attributes.get("_datadog", {})
-        dd_json_data = dd_payload.get("stringValue", dd_payload.get("Value", r"{}"))
+        # SQS uses dataType and binaryValue/stringValue
+        # SNS uses Type and Value
+        dd_json_data_type = dd_payload.get("Type", dd_payload.get("dataType", ""))
+        if dd_json_data_type == "Binary":
+            dd_json_data = dd_payload.get(
+                "binaryValue",
+                dd_payload.get("Value", r"{}"),
+            )
+            dd_json_data = base64.b64decode(dd_json_data)
+        elif dd_json_data_type == "String":
+            dd_json_data = dd_payload.get(
+                "stringValue",
+                dd_payload.get("Value", r"{}"),
+            )
+        else:
+            raise Exception(
```

Comment:
Is this error caught anywhere? Why raise here when we previously didn't?

If for some reason the `_datadog` attribute isn't string or binary, we don't want to throw an error and halt the customer function execution.

Reply:
I now see why we wouldn't want to stop the execution if its not a string or binary. I opted for making this exception a logger.warn as I think it would still be useful for the customer to know that their message attribute type is not supported by our library.

Reply:
Nice, I think that's a good call